kube-configmap.go

```diff
@@ -0,0 +1,365 @@
+package ipam
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"sync"
+
+	corev1 "k8s.io/api/core/v1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type kubeConfigMap struct {
+	client       client.Client
+	configMapKey types.NamespacedName
+	lock         sync.Mutex
+}
+
+func NewKubeConfigMap(ctx context.Context, client client.Client, namespacedName types.NamespacedName) (Storage, error) {
+	kcm := &kubeConfigMap{
+		client:       client,
+		configMapKey: namespacedName,
+	}
+
+	if err := kcm.CreateNamespace(ctx, defaultNamespace); err != nil {
```

Comment:
Isnt it required to check for a apierrors.IsNotFound ?

Reply:
That case has already been handled in the loadConfigMap and storeConfigMap functions.